### PR TITLE
gravel: fix bugs on bootstrap and join

### LINKED ISSUE #309: gravel: initial node requires aquarium restart after bootstrap before join is permitted
I'm not entirely sure how to describe the bug, aside from what happens: after we bootstrap the initial node, join attempts from another node result in a closed websocket with an invalid status code. Restarting Aquarium on the initial node makes everything okay without an itch.

At this point I have no concrete ideas about what might be wrong, but I suspect it has something to do with the node mgr / connection mgr rejecting a connection depending on state. Some state must be wrong somewhere, or we're simply not starting something.

### PR DESCRIPTION
**Pull request by jecluis**
Resolves #309 

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/gravel/controllers/bootstrap.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ async def _do_bootstrap(self) -> None:
         self.stage = BootstrapStage.RUNNING
 
         def progress_cb(percent: int) -> None:
-            logger.debug(f"bootstrap > {percent}")
+            logger.debug(f"bootstrap > {percent}%")
```
Comment (mgfritch) [inline src/gravel/controllers/bootstrap.py:113]:
```suggestion
            logger.debug(f"bootstrap {percent}%")
```
(nit)

File: src/gravel/controllers/nodes/mgr.py [hunk 2]
```diff
@@ -467,6 +431,31 @@ def _load_token(self) -> Optional[str]:
...
 
+    def _save_token(self, should_exist: bool = True) -> None:
+        tokenfile: Path = self._get_node_file("token")
+
+        # this check could be a single assert, but it's important to know which
+        # path failed.
+        if should_exist:
+            assert tokenfile.exists()
+        else:
+            assert not tokenfile.exists()
```
Comment (mgfritch) [inline src/gravel/controllers/nodes/mgr.py:442]: I'm confused, do we really need to assert this?

**Review by mgfritch — APPROVED** [review]
couple nits, but lgtm!

File: src/gravel/controllers/nodes/mgr.py (hunk 2 above)
Comment (jecluis) [reply src/gravel/controllers/nodes/mgr.py:442]: We don't need to assert it, but it sure comes handy to ensure we are getting the right state at the right time from the caller's PoV. The alternative would be having the caller perform the asserts first, but that would require them to know where the file sits and hence increase boilerplate quite a bit.

**Pull request merged**